agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -2079,6 +2079,13 @@ bool slave_thread::handle_cmdu_ap_manager_message(Socket *sd,
         notification_out->iface_mac()  = hostap_params.iface_mac;
         notification_out->client_mac() = notification_in->params().mac;
         notification_out->bssid()      = notification_in->params().bssid;
+        if (!notification_in->params().association_frame) {
+            LOG(DEBUG) << "no association frame";
```

Comment:
I don't think it useless since if the association frame is empty it means something is wrong, and it will be printed only if something wrong..

Reply:
So change to `LOG_IF(!notification_in->params().association_frame, ERROR) << "no association frame";`

Reply:
No, it's literally useless, because `association_frame` is an array, not a pointer:

```cpp
#include <iostream>

struct s {
    char foo[10];
};

int main()
{
    s s1;
    s1.foo[0] = '\0';
    if (!s1.foo)
        std::cout << "foo is empty" << std::endl;
    else
        std::cout << "foo is not empty" << std::endl;
}
```

The code above will print "foo is not empty".

It *could* be useful to say `if (notification_in->params().association_frame[0] != '\0')`.

However, as I wrote elsewhere, it should be a binary buffer, not a hex string. IMO.